### PR TITLE
New eWallets API

### PR DESCRIPTION
**Pull request by glendaesutanto**
- `POST` Create eWallet charge
https://developers.xendit.co/api-reference/#create-ewallet-charge
- `GET` Get eWallet charge status
https://developers.xendit.co/api-reference/#get-ewallet-charge-status

### CODE REVIEW

File: ewallet.go [hunk 1]
```diff
@@ -26,3 +26,42 @@ type EWallet struct {
 	Created              *time.Time      `json:"created,omitempty"`
 	EWalletTransactionID string          `json:"e_wallet_transaction_id,omitempty"`
 }
+
+type BasketItem struct {
```
Comment (stanleynguyen) [inline ewallet.go:30]: `EwalletBasketItem` instead?

Comment (glendaesutanto) [reply ewallet.go]: done

File: ewallet/client.go [hunk 2]
```diff
@@ -111,3 +163,71 @@ func (c *Client) GetPaymentStatusWithContext(ctx context.Context, data *GetPayme
 
 	return response, nil
 }
+
+// CreateEWalletCharge creates new e-wallet charge
+func (c *Client) CreateEWalletCharge(data *CreateEWalletChargeParams) (*xendit.EWalletCharge, *xendit.Error) {
+	return c.CreateEWalletChargeWithContext(context.Background(), data)
+}
```
Comment (bxcodec) [inline ewallet/client.go:170]: Do we will still need this?

Comment (bxcodec) [reply ewallet/client.go:170]: I mean, why we will still support the function without the `context` param?

Comment (glendaesutanto) [reply ewallet/client.go:170]: Oh I'm sorry, this is my first time using Go actually so I'm not really familiar and I'm still learning it. For this one, I just tried to follow the pattern that occurs in that file. What is the best practice for this one?

**Review by stanleynguyen — APPROVED** [review]

File: ewallet/client.go (hunk 2 above)
Comment (stanleynguyen) [reply ewallet/client.go:170]: @bxcodec We have always been supporting both option:
- a no-frill one without context
- an explicitly context-requiring one

**Pull request merged**